### PR TITLE
Feature. Template TS: integrate react-navigation

### PR DESCRIPTION
**Pull request by vshkl**
- Integrate `react-navigation`
- Add example HomeScreen

### CODE REVIEW

File: template-js/template/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { AppRegistry } from 'react-native'
 import 'react-native-gesture-handler'
```
Comment (szholdiyarov) [inline template-js/template/index.js:1]: why we need to import the whole package here?

File: template-ts/template/src/navigation/MainNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { NavigationContainer } from '@react-navigation/native'
+import { createStackNavigator } from '@react-navigation/stack'
+
+import HomeScreen from '../screens/HomeScreen'
+
+type MainStackParamList = {
+  HomeScreen: undefined,
```
Comment (szholdiyarov) [inline template-ts/template/src/navigation/MainNavigator.tsx:8]: If you want to specify the "empty" state Implicitly, I think it is better to use null here.

File: template-js/template/index.js (hunk 1 above)
Comment (vshkl) [reply template-js/template/index.js:1]: It's required by the library

File: template-ts/template/src/navigation/MainNavigator.tsx (hunk 2 above)
Comment (vshkl) [reply template-ts/template/src/navigation/MainNavigator.tsx:8]: It's from the lib docs

File: template-js/template/index.js (hunk 1 above)
Comment (vshkl) [reply template-js/template/index.js:1]: https://reactnavigation.org/docs/getting-started#installing-dependencies-into-a-bare-react-native-project

File: template-ts/template/src/navigation/MainNavigator.tsx (hunk 2 above)
Comment (vshkl) [reply template-ts/template/src/navigation/MainNavigator.tsx:8]: https://reactnavigation.org/docs/typescript#type-checking-the-navigator

Comment (vshkl) [reply template-ts/template/src/navigation/MainNavigator.tsx:8]: Also, `null` is kinda for empty value while `undefined` is for the absence of any value.

**Comment by vshkl** [comment]
>I left a few minor comments and would like to clarify why you would like to divide the screen into two separate files container and index.js?

Well... no particular reason. For me it's just clearer that container is a container. I'll change it in the next PR, deal?

**Pull request merged**